client/jetpack-cloud/sections/partner-portal/primary/payment-method-add/index.tsx

```diff
@@ -11,10 +17,159 @@ import Main from 'calypso/components/main';
+import type { Purchase } from 'calypso/lib/purchases/types';
+import type { TranslateResult } from 'i18n-calypso';
+import FormInputCheckbox from 'calypso/components/forms/form-checkbox';
+import FormLabel from 'calypso/components/forms/form-label';
+import CreditCardLoading from 'calypso/jetpack-cloud/sections/partner-portal/credit-card-fields/credit-card-loading';
+import DeletePaymentMethodDialog from 'calypso/jetpack-cloud/sections/partner-portal/delete-payment-method-dialog';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
 
-export default function PaymentMethodAdd(): ReactElement {
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+function onPaymentSelectComplete( {
+	successCallback,
+	translate,
+	showSuccessMessage,
+	purchase,
+}: {
+	successCallback: () => void;
+	translate: ReturnType< typeof useTranslate >;
+	showSuccessMessage: ( message: string | TranslateResult ) => void;
+	purchase?: Purchase | undefined;
+} ) {
+	if ( purchase ) {
+		showSuccessMessage( translate( 'Your payment method has been set.' ) );
+	} else {
+		showSuccessMessage( translate( 'Your payment method has been added successfully.' ) );
+	}
+	successCallback();
+}
+
+function CurrentPaymentMethodNotAvailableNotice( {
+	purchase,
+}: {
+	purchase: Purchase;
+} ): JSX.Element | null {
+	const translate = useTranslate();
+	const moment = useLocalizedMoment();
+	const noticeProps: Record< string, boolean | string | number | TranslateResult > = {
+		showDismiss: false,
+	};
+
+	if ( purchase.payment.creditCard && creditCardHasAlreadyExpired( purchase ) ) {
+		noticeProps.text = translate(
+			'Your %(cardType)s ending in %(cardNumber)d expired %(cardExpiry)s.',
+			{
+				args: {
+					cardType: purchase.payment.creditCard.type.toUpperCase(),
+					cardNumber: parseInt( purchase.payment.creditCard.number, 10 ),
+					cardExpiry: moment( purchase.payment.creditCard.expiryDate, 'MM/YY' ).format(
+						'MMMM YYYY'
+					),
+				},
+			}
+		);
+		return <Notice { ...noticeProps } />;
+	}
+
+	return null;
+}
+
+function PaymentMethodAdd(): ReactElement {
+	const purchase = undefined;
+	const dispatch = useDispatch();
 	const translate = useTranslate();
+	const { isStripeLoading, stripeLoadingError, stripeConfiguration, stripe } = useStripe();
+	const detailsId = getQueryArg( window.location.href, 'details_id' );
+	const storedCards = useSelector( getAllStoredCards );
+	const currentCard = storedCards.find( ( card ) => detailsId === card.stored_details_id );
+	const isEditing = detailsId && currentCard;
+
+	useEffect( () => {
+		if ( detailsId && ! currentCard ) {
+			page.redirect(
+				removeQueryArgs( window.location.pathname + window.location.search, 'details_id' )
+			);
+		}
+	}, [] );
+
+	const stripeMethod = useCreateStoredCreditCard( {
+		isStripeLoading,
+		stripeLoadingError,
+		stripeConfiguration,
+		stripe,
+		activePayButtonText: translate( 'Save payment method' ) as string,
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **16** times:
`translate( 'Saved payment method' )` *ES Score: 8*